### PR TITLE
BFW-2365: Disable networking in blue/redscreen

### PR DESCRIPTION
**Pull request by vorner**
Because it's weird if the printer can't be controlled by GUI and acts
„dead“, while still communicating over network (accepting requests,
letting gcode be uploaded, and eventually it would even allow one to
start printing).

### CODE REVIEW

**Review by espr14 — APPROVED** [review]

**Review by espr14 — CHANGES_REQUESTED** [review]

**Review by DRracer — APPROVED** [review]

File: src/main.cpp [hunk 1]
```diff
@@ -312,6 +312,27 @@ int main(void) {
...
     defaultTaskHandle = osThreadCreate(osThread(defaultTask), NULL);
 
+#ifdef BUDDY_ENABLE_WUI
+    /*
+     * Checking this first, before starting the GUI thread. The GUI thread
+     * resets/consumes the dump as a side effect.
+     */
+    bool block_networking = false;
+
+    if (dump_in_xflash_is_valid() && !dump_in_xflash_is_displayed()) {
```
[the remark this answers is not in the document]
Comment (DRracer) [reply src/main.cpp:322]: are you sure the `w25x_init()` is not being done elsewhere?

[the remark this answers is not in the document]
Comment (espr14) [reply src/main.cpp:322]: `if (w25x_init())` and `w25x_init()` could be different. No documentation available so I cannot tell if we can omit it.

[the remark this answers is not in the document]
Comment (DRracer) [reply src/main.cpp:322]: In this case, if the w25x chip is not inited (`w25x_init()` returns false) at the beginning of main.cpp, the code will fail much earlier, so the changes introduced by @vorner won't be even reached IMO.

[the remark this answers is not in the document]
Comment (vorner) [reply src/main.cpp:322]: I didn't feel like initializing it yet once more… but if you think it makes sense, I can do it. Or put it together at the top.

**Review by espr14 — APPROVED** [review]

**Pull request merged**